modules/dhcp/api/gr_dhcp.h

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2025 Anthony Harivel
+
+#ifndef _GR_DHCP
+#define _GR_DHCP
+
+#include <gr_api.h>
+#include <gr_infra.h>
+#include <gr_net_types.h>
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum dhcp_state : uint8_t {
+	DHCP_STATE_INIT = 0,
+	DHCP_STATE_SELECTING,
+	DHCP_STATE_REQUESTING,
+	DHCP_STATE_BOUND,
+	DHCP_STATE_RENEWING,
+	DHCP_STATE_REBINDING,
+};
+
+struct gr_dhcp_status {
+	uint16_t iface_id;
+	enum dhcp_state state;
+	ip4_addr_t server_ip;
+	ip4_addr_t assigned_ip;
+	uint32_t lease_time;
+	uint32_t renewal_time; // T1
+	uint32_t rebind_time; // T2
+};
+
+struct gr_dhcp_list_resp {
+	uint16_t n_dhcp;
+	struct gr_dhcp_status dhcp[];
+};
+
+struct gr_dhcp_start_req {
+	uint16_t iface_id;
+};
+
+struct gr_dhcp_stop_req {
+	uint16_t iface_id;
+};
```

Comment:
Could you follow the same model than other headers? With interleaved #define GR_DHCP_LIST and their related structs.